armi/reactor/blocks.py

```diff
@@ -1580,61 +1580,85 @@ def setPinPowers(
+            Defaults to False
+
+        powerKeySuffix: str, optional
+            Must be either an empty string, :py:const:`NEUTRON <armi.physics.neutronics.const.NEUTRON>`,
+            or :py:const:`GAMMA <armi.physics.neutronics.const.GAMMA>`. Defaults to empty
+            string.
+
+        Notes
+        -----
+        This method can handle assembly rotations by using the *pinLocation* parameter.
         """
-        self.p.pinPowers = numpy.zeros(numPins)
-        self.p["linPowByPin" + powerKeySuffix] = numpy.zeros(numPins)
+        powerKey = f"linPowByPin{powerKeySuffix}"
+        self.p[powerKey] = numpy.zeros(numPins)
+
+        # Loop through rings
         j0 = jmax[imax - 1] / 6
         pinNum = 0
-        for i in range(imax):  # loop through rings
-            for j in range(jmax[i]):  # loop through positions in ring i
+        for i in range(imax):
+            # Loop through positions in ring i
+            for j in range(jmax[i]):
                 if removeSixCornerPins and i == imax - 1 and math.fmod(j, j0) == 0.0:
                     linPow = 0.0
                 else:
+                    # TODO: This appears to need fixing to account for blocks in fueled
+                    # TODO: assemblies that contain elevations with pins but no fuel,
+                    # TODO: such as for an axial shield. These blocks should still have
+                    # TODO: powers that are eligible for rotation.
                     if self.hasFlags(Flags.FUEL):
-                        # -1 to map from pinLocations to list index
+                        # -1 is needed in order to map from pinLocations to list index
                         pinLoc = self.p.pinLocation[pinNum] - 1
                     else:
                         pinLoc = pinNum
                     linPow = powers[pinLoc]
-                self.p.pinPowers[pinNum] = linPow
-                self.p["linPowByPin" + powerKeySuffix][pinNum] = linPow
+                self.p[powerKey][pinNum] = linPow
                 pinNum += 1
 
-        if powerKeySuffix == GAMMA:
-            self.p.pinPowersGamma = self.p.pinPowers
-        elif powerKeySuffix == NEUTRON:
-            self.p.pinPowersNeutron = self.p.pinPowers
-
-        if gamma:
-            self.p.pinPowers = self.p.pinPowersNeutron + self.p.pinPowersGamma
-        else:
-            self.p.pinPowers = self.p.pinPowersNeutron
+        # If using the *powerKeySuffix* parameter, we also need to set total power, which
+        # is sum of neutron and gamma powers. We assume that a solo gamma calculation
+        # to set total power does not make sense.
+        if powerKeySuffix:
+            if powerKeySuffix == GAMMA:
+                if self.p[f"linPowByPin{NEUTRON}"] is None:
+                    msg = (
+                        "Neutron power has not been set yet. Cannot set total power for "
+                        f"{self}."
+                    )
+                    raise UnboundLocalError(msg)
```

Comment:
I believe this is why your code coverage went down for this PR.

You added a new special exception, but you aren't testing for it in a unit test.